### PR TITLE
rn-oauth2: initial skeleton for the lib

### PR DESCRIPTION
**Pull request by martinmilev**
Added basic functionality.
It's a working example how the lib would look like, but fetching credentials
and other functionalities should be added in future commits.

### CODE REVIEW

**Review by gadjevb — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        if (tokenRequest.hasOwnProperty(key)) {
+          s.push(
+             encodeURIComponent(key) +
+              '=' +
+              encodeURIComponent(tokenRequest[key])
+          )
+        }
+      }
+      const targetUrl = s.join('&')
+      console.warn('targetUrl: ', targetUrl)
```
Comment (mgenov) [inline index.js]: missed warning

File: index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        },
+        credentials: {}
+      }
+    }
+  }
+
+  static logout() {
+    Linking.openURL(LOGOUT + callback)
+      .then(response => {
+        return false
```
Comment (mgenov) [inline index.js]: return is not required here as it's in a promise

File: index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      })
+      .catch(error => {
+        console.log('logout error: ', error)
```
Comment (mgenov) [inline index.js]: missed log

File: README.md [hunk 4]
```diff
@@ -1,2 +1,10 @@
 # react-native-oauth2
 OAuth2 Package for RN
+
+## Usage
+```javascript
+import Oauth2 from 'react-native-oauth2'
+
+Oauth.authenticate()
```
Comment (mgenov) [inline README.md]: Promisee should be returned from both as these are async operations.

**Review by mgenov — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -1,2 +1,10 @@
 # react-native-oauth2
 OAuth2 Package for RN
+
+## Usage
+```javascript
+import Oauth2 from 'react-native-oauth2'
```
Comment (mgenov) [inline README.md]: OAuth2

It maybe should be something like: 

```javascript
let client = OAuth2.newClient(clientId, secret, authUrl, tokenUrl) 
client.authenticate().then(response => {
  // authentication was performed 
}).catch(error => {
  // oops, got error during signing in, which could be shown to the user
})
```

Comment (mgenov) [reply README.md]: Last comment was updated with example usage which need to be considered for the design.

File: index.js (hunk 1 above)
Comment (martinmilev) [reply index.js]: removed.

File: README.md (hunk 4 above)
Comment (martinmilev) [reply README.md]: removed.

File: index.js (hunk 2 above)
Comment (martinmilev) [reply index.js]: fixed

File: index.js (hunk 3 above)
Comment (martinmilev) [reply index.js]: removed.

File: README.md (hunk 5 above)
Comment (martinmilev) [reply README.md]: fixed

File: README.md [hunk 6]
```diff
@@ -1,2 +1,15 @@
 # react-native-oauth2
 OAuth2 Package for RN
+
+## Usage
+```javascript
+  import Oauth from 'react-native-oauth2'
+
+  const client = new Oauth(APP_ID, APP_CALLBACK, ACCOUNTS_URL, ACCOUNTS_URL)
```
Comment (mgenov) [inline README.md]: Use AUTH_URL and TOKEN_URL instead of ACCOUNTS_URL

**Review by mgenov — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (martinmilev) [reply README.md]: fixed.

**Review by mgenov — APPROVED** [review]
LGTM !

**Pull request merged**